### ISSUE TITLE
Change placeholder text color for Select

### ISSUE THREAD

**Issue by haase1020** [spear-ai/ui#259]
The `placeholder-shown` data attribute should have the text color of `text-neutral-11` to meet color contrast..
https://github.com/spear-ai/ui/blob/a062066d9ee1268674b0e59b97c24049888fedfe/packages/ui/src/components/select.tsx#L134

**Comment by psirenny** [comment]
This is an example where I think it's a mistake to follow the contrast color report. IMO those are warnings that should be followed up with an investigation rather than hard and fast rules. It's OK if we don't get a 100%.

Radix Colors says to use color step 9 for placeholder values.
<img width="739" alt="Screenshot 2024-05-25 at 8 43 02 AM" src="https://github.com/spear-ai/ui/assets/463178/b6bb5d25-8ee8-4cf6-8660-60298c85ba24">

However, Radix breaks this rule in the playground. Instead, they consistently use `step-10`. I suspect they just failed to update their color documentation after a recent update.

In my opinion, `step-11` has too much contrast for a placeholder value and so `step-10` is better. It might technically be a little harder to see, but on net it's probably worse to assign placeholders a color that will be confused with real values.

As always, @eakbulut can override me. The only thing that I want to insist on is that we don't dogmatically follow these accessibility scores. I've seen lots of great (accessible) websites break these rules all the time.

**Comment by haase1020** [comment]
I do think following the guidelines lighthouse follows is important for accessibility issues, as lighthouse follows the Web Content Accessibility Guidelines (WCAG). 

Here is the WCAG section on placeholders: https://www.w3.org/WAI/tutorials/forms/instructions/#placeholder-text
